api/src/main/java/com/xwiki/googleapps/DriveDocMetadata.java

```diff
@@ -44,4 +47,157 @@
      * URL to pull from in order to fetch the document.
      */
     public String exportLink;
+
+    /**
+     * URL to use to show an embedded view.
+     */
+    public String embedLink;
+
+    /**
+     * A stringified version number.
+     */
+    public String version;
+
+    /**
+     * The name of the file in case it is an uploaded file.
+     */
+    public String fileName;
+
+    /**
+     * The email-address of the user with which this document's connection was created.
+     */
+    public String user;
+
+    /**
+     * A list of export possibilities.
+     */
+    public List<ExportAlternative> exportLinksAlternatives = new LinkedList<>();
```

Comment:
Getters are needed for velocity properties to work, it seems. Am I wrong?

Reply:
If that's so then I would add setters too and make the fields private. I find it very confusing to have a class with public fields and only getters (and no comment to explain it). Plus, this class doesn't seem to be used only by scripts (Velocity) so I'm not convinced it's OK to justify design decisions because objects of this class might be used in scripts.

Reply:
Indeed this inner class is not used other than inside the enclosing or velocity.

Are you suggesting to use a map instead?
I thought it makes it more readable to use a class with designated attributes.
Indeed the fields need to not be public at all.

Reply:
> Are you suggesting to use a map instead?

No, using typed data is better. My suggestion is to make it a Java bean: private fields with public getters and setters. If it weren't for Velocity, I would have kept only the fields, but since we need the getters for Velocity then I suggest you add the setters also. Otherwise the code looks really strange, with public fields and public getters and no setters.

Reply:
Done.